analyzer/block/block.go

```diff
@@ -425,13 +466,20 @@ func NewAnalyzer(
 	if batchSize == 0 {
 		batchSize = defaultBatchSize
 	}
-	return &blockBasedAnalyzer{
+	a := &blockBasedAnalyzer{
 		blockRange:   blockRange,
 		batchSize:    batchSize,
 		analyzerName: name,
 		processor:    processor,
 		target:       target,
 		logger:       logger.With("analyzer", name, "mode", mode),
-		slowSync:     mode == analyzer.SlowSyncMode,
-	}, nil
+		queueLengthMetric: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: fmt.Sprintf("%s_queue_length", name),
+			Help: fmt.Sprintf("number of blocks left to process for the %s analyzer", name),
+		}),
```

Comment:
Request for small refactor: Can you predefine this metric in `metrics/analysis.go`? Having it there is nice because
- it keeps metric definitions in one place, making it easier to understand what our prometheus endpoint exposes
- it defines helper funcs that let you create metrics with _type-checked_ labels; see e.g. `DatabaseLatencies(...)`
- (more theoretical) it makes it easier to mock out the entirety of prometheus, should we ever want to

So the idea is to
1. define a new gauge-typed `queueLength` field inside `AnalysisMetrics`, with `analyzer` being one of the labels, possibly the only one
2. use this predefined metric from block.go but also item.go (so neither will need a `queueLengthMetric` field any more)

Reply:
Done!